### ISSUE TITLE
tmpdir - case sensitive dir names prevent certain parametrization on windows

### ISSUE THREAD

**Issue by aostr** [pytest-dev/pytest#2323]
Here's a simple example of a test which can't be executed on windows using tmpdir fixture.
```
import pytest

@pytest.mark.parametrize("param1", ["a", "A"])
def test_tmpdir_fail_windows(tmpdir, param1):
   print(tmpdir)

```

In this case the tmpdir names are `test_tmpdir_fail_windows_a_0` and `test_tmpdir_fail_windows_A_0` which fails badly on windows where the directory names are not case-sensitive.

Here's what I see on the console
```
(test_env1) c:\tmp>pytest test_tmpdir_fail.py -vs
============================= test session starts =============================
platform win32 -- Python 3.5.1, pytest-3.0.7, py-1.4.33, pluggy-0.4.0 -- c:\tmp\test_env1\scripts\python.e
cachedir: .cache
rootdir: c:\tmp, inifile:
collected 2 items

test_tmpdir_fail.py::test_tmpdir_fail_windows[a] C:\Users\aostr\AppData\Local\Temp\pytest-of-aostr\pytest-
r_fail_windows_a_0
PASSED
test_tmpdir_fail.py::test_tmpdir_fail_windows[A] ERROR

=================================== ERRORS ====================================
________________ ERROR at setup of test_tmpdir_fail_windows[A] ________________
```
followed by a huge traceback.

Here's my `pip list` if it changes anything:
```
(test_env1) c:\tmp>pip list
DEPRECATION: The default format will switch to columns in the fut
 format=(legacy|columns) in your pip.conf under the [list] sectio
appdirs (1.4.3)
check-manifest (0.35)
colorama (0.3.7)
devpi-client (2.7.0)
devpi-common (3.0.1)
packaging (16.8)
pip (9.0.1)
pkginfo (1.4.1)
pluggy (0.4.0)
py (1.4.33)
pyparsing (2.2.0)
pytest (3.0.7)
requests (2.13.0)
setuptools (28.8.0)
six (1.10.0)
tox (2.6.0)
virtualenv (15.1.0)
wheel (0.29.0)
```

I would guess that the issue can be reproduced on any windows machine.
My setup in this case is Windows 8.1 64-bit, pytest-3.0.7, python 3.5.1 (32-bit interpreter) - if this info brings anything to the problem :)

**Comment by RonnyPfannschmidt** [comment]
please provide the full traceback (im not on windows)

**Comment by aostr** [comment]
The traceback below.

Would it make sense to just call the `.lowercase()` method on the dir name somewhere in the builtin tmpdir fixture and make use of the numbered directory names? I'm not sure if this would solve the problem for everybody though. It would work for me.
```
============================= test session starts =============================
platform win32 -- Python 3.5.1, pytest-3.0.7, py-1.4.33, pluggy-0.4.0 -- c:\tmp\test_env1\scripts\python.exe
cachedir: .cache
rootdir: c:\tmp, inifile:
collecting ... collected 2 items

test_tmpdir_fail.py::test_tmpdir_fail_windows[a] C:\Users\aostr\AppData\Local\Temp\pytest-of-aostr\pytest-312\test_tmpdir_fail_windows_a_0
PASSED
test_tmpdir_fail.py::test_tmpdir_fail_windows[A] ERROR

=================================== ERRORS ====================================
________________ ERROR at setup of test_tmpdir_fail_windows[A] ________________

self = <py._error.ErrorMaker object at 0x00954350>
func = <built-in function mkdir>
args = ('C:\\Users\\aostr\\AppData\\Local\\Temp\\pytest-of-aostr\\pytest-312\\test_tmpdir_fail_windows_A_0',)
kwargs = {}, __tracebackhide__ = False, cls = <class 'py.error.EEXIST'>
value = FileExistsError(17, 'Cannot create a file when that file already exists')
tb = <traceback object at 0x02DFD788>, errno = 17

    def checked_call(self, func, *args, **kwargs):
        """ call a function and raise an errno-exception if applicable. """
        __tracebackhide__ = True
        try:
>           return func(*args, **kwargs)
E           FileExistsError: [WinError 183] Cannot create a file when that file already exists: 'C:\\Users\\aostr\\AppData\\Local\\Temp\\pytest-of-aostr\\pytest-312\\test_tmpdir_fail_windows_A_0'

test_env1\lib\site-packages\py\_error.py:65: FileExistsError

During handling of the above exception, another exception occurred:

self = <CallInfo when='setup' exception: [File exists]: mkdir('C:\\Users\\aostr\\AppData\\Local\\Temp\\pytest-of-aostr\\pytest-312\\test_tmpdir_fail_windows_A_0',)>
func = <function call_runtest_hook.<locals>.<lambda> at 0x02D06540>
when = 'setup'

    def __init__(self, func, when):
        #: context of invocation: one of "setup", "call",
        #: "teardown", "memocollect"
        self.when = when
        self.start = time()
        try:
>           self.result = func()

test_env1\lib\site-packages\_pytest\runner.py:163: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
test_env1\lib\site-packages\_pytest\runner.py:151: in <lambda>
    return CallInfo(lambda: ihook(item=item, **kwds), when=when)
test_env1\lib\site-packages\_pytest\vendored_packages\pluggy.py:745: in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
test_env1\lib\site-packages\_pytest\vendored_packages\pluggy.py:339: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
test_env1\lib\site-packages\_pytest\vendored_packages\pluggy.py:334: in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
test_env1\lib\site-packages\_pytest\vendored_packages\pluggy.py:613: in execute
    return _wrapped_call(hook_impl.function(*args), self.execute)
test_env1\lib\site-packages\_pytest\vendored_packages\pluggy.py:254: in _wrapped_call
    return call_outcome.get_result()
test_env1\lib\site-packages\_pytest\vendored_packages\pluggy.py:279: in get_result
    raise ex[1].with_traceback(ex[2])
test_env1\lib\site-packages\_pytest\vendored_packages\pluggy.py:265: in __init__
    self.result = func()
test_env1\lib\site-packages\_pytest\vendored_packages\pluggy.py:614: in execute
    res = hook_impl.function(*args)
test_env1\lib\site-packages\_pytest\runner.py:100: in pytest_runtest_setup
    item.session._setupstate.prepare(item)
test_env1\lib\site-packages\_pytest\runner.py:453: in prepare
    col.setup()
test_env1\lib\site-packages\_pytest\python.py:1578: in setup
    fixtures.fillfixtures(self)
test_env1\lib\site-packages\_pytest\fixtures.py:253: in fillfixtures
    request._fillfixtures()
test_env1\lib\site-packages\_pytest\fixtures.py:398: in _fillfixtures
    item.funcargs[argname] = self.getfixturevalue(argname)
test_env1\lib\site-packages\_pytest\fixtures.py:440: in getfixturevalue
    return self._get_active_fixturedef(argname).cached_result[0]
test_env1\lib\site-packages\_pytest\fixtures.py:465: in _get_active_fixturedef
    result = self._getfixturevalue(fixturedef)
test_env1\lib\site-packages\_pytest\fixtures.py:530: in _getfixturevalue
    val = fixturedef.execute(request=subrequest)
test_env1\lib\site-packages\_pytest\fixtures.py:786: in execute
    return ihook.pytest_fixture_setup(fixturedef=self, request=request)
test_env1\lib\site-packages\_pytest\vendored_packages\pluggy.py:745: in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
test_env1\lib\site-packages\_pytest\vendored_packages\pluggy.py:339: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
test_env1\lib\site-packages\_pytest\vendored_packages\pluggy.py:334: in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
test_env1\lib\site-packages\_pytest\vendored_packages\pluggy.py:613: in execute
    return _wrapped_call(hook_impl.function(*args), self.execute)
test_env1\lib\site-packages\_pytest\vendored_packages\pluggy.py:254: in _wrapped_call
    return call_outcome.get_result()
test_env1\lib\site-packages\_pytest\vendored_packages\pluggy.py:279: in get_result
    raise ex[1].with_traceback(ex[2])
test_env1\lib\site-packages\_pytest\vendored_packages\pluggy.py:265: in __init__
    self.result = func()
test_env1\lib\site-packages\_pytest\vendored_packages\pluggy.py:614: in execute
    res = hook_impl.function(*args)
test_env1\lib\site-packages\_pytest\fixtures.py:816: in pytest_fixture_setup
    result = call_fixture_func(fixturefunc, request, kwargs)
test_env1\lib\site-packages\_pytest\fixtures.py:721: in call_fixture_func
    res = fixturefunc(**kwargs)
test_env1\lib\site-packages\_pytest\tmpdir.py:123: in tmpdir
    x = tmpdir_factory.mktemp(name, numbered=True)
test_env1\lib\site-packages\_pytest\tmpdir.py:39: in mktemp
    keep=0, rootdir=basetemp, lock_timeout=None)
test_env1\lib\site-packages\py\_path\local.py:825: in make_numbered_dir
    udir = rootdir.mkdir(prefix + str(maxnum+1))
test_env1\lib\site-packages\py\_path\local.py:459: in mkdir
    py.error.checked_call(os.mkdir, fspath(p))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <py._error.ErrorMaker object at 0x00954350>
func = <built-in function mkdir>
args = ('C:\\Users\\aostr\\AppData\\Local\\Temp\\pytest-of-aostr\\pytest-312\\test_tmpdir_fail_windows_A_0',)
kwargs = {}, __tracebackhide__ = False, cls = <class 'py.error.EEXIST'>
value = FileExistsError(17, 'Cannot create a file when that file already exists')
tb = <traceback object at 0x02DFD788>, errno = 17

    def checked_call(self, func, *args, **kwargs):
        """ call a function and raise an errno-exception if applicable. """
        __tracebackhide__ = True
        try:
            return func(*args, **kwargs)
        except self.Error:
            raise
        except (OSError, EnvironmentError):
            cls, value, tb = sys.exc_info()
            if not hasattr(value, 'errno'):
                raise
            __tracebackhide__ = False
            errno = value.errno
            try:
                if not isinstance(value, WindowsError):
                    raise NameError
            except NameError:
                # we are not on Windows, or we got a proper OSError
                cls = self._geterrnoclass(errno)
            else:
                try:
                    cls = self._geterrnoclass(_winerrnomap[errno])
                except KeyError:
                    raise value
>           raise cls("%s%r" % (func.__name__, args))
E           py.error.EEXIST: [File exists]: mkdir('C:\\Users\\aostr\\AppData\\Local\\Temp\\pytest-of-aostr\\pytest-312\\test_tmpdir_fail_windows_A_0',)

test_env1\lib\site-packages\py\_error.py:85: EEXIST
====================== 1 passed, 1 error in 0.46 seconds ======================
```

**Comment by RonnyPfannschmidt** [comment]
ok, thanks, this is indeed a pylib bug, i will report it at the correct location and keep this issue open to track it

**Comment by aostr** [comment]
Awesome, thanks Ronny! Cheers.

**Comment by crisidev** [comment]
This is also happening on OSX when using a case-sensitive file system.

**Comment by nicoddemus** [comment]
Should be fixed with `py-1.4.34`
